web-app/HeimScripts/heatmap/run.r

```diff
@@ -1,29 +1,90 @@
 library(gplots)
 
 
+dataset1color <- "coral3"
+dataset2color <- "chartreuse3"
+labelColumns <- c("Row.Label","Bio.marker")
+
+#Input expected 1 or 2 dataframes with columns: Row.Label, Bio.marker, ASSAY_0001 ASSAY_0002 ...
 main <- function(){
-    dataset <- loaded_variables[[1]] #dataframe with columns: Row.Label, Bio.marker, ASSAY_0001 ASSAY_0002 ...
+  datasets <- parseInput(loaded_variables) #this will just make sure we have either 1 or 2 dataframes in a list 
+  measurements <- extractMeasurements(datasets) #extract the numeric part - as a numeric matrix is needed for the heatmap.2 function
+  humanReadableNames <-extractNames(datasets)
+  measurements <- assignNames(measurements,humanReadableNames) #combine label with Biomarker(if present) and assign as a name to be displayed on the heatmap
+  measurements <- transform(measurements) #log2 transform the numerical matrix with measurements
+  grouping <- extractGrouping(datasets) #Grouping determines coloring of the samples - one color for dataset1 and a different one for dataset2
+  makeHeatmap(measurements,grouping) # plot the heatmap and save it in the working directory as a .png file
+}
 
-    measurements  <- subset(dataset,select=-c(Row.Label,Bio.marker)) # this will select all columns other than Row.Label,Bio.marker columns
-    measurements  <- data.matrix(measurements)
+parseInput <- function(variables){
+  datasets <- list()
+  varLength <- length(variables)
+  if(varLength > 1){
+    datasets <- list(dataset1 = variables[[1]], dataset2 = variables[[2]])  
+  }
+  else if(varLength == 1){
+    datasets <- list(dataset1 = variables[[1]])
+  }
+  return(datasets)
+}
 
```

Comment:
Is this function extracting maximally 2 elements from an existing list? It seems a lot of code for doing this.